semtypes/src/main/java/io/ballerina/types/MappingAlternative.java

```diff
@@ -0,0 +1,74 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+package io.ballerina.types;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ *  Represents MappingAlternative record in core.bal.
+ *
+ * @since 3.0.0
+ */
+public class MappingAlternative {
+    SemType semType;
+    MappingAtomicType[] pos;
+    MappingAtomicType[] neg;
+
+    private MappingAlternative(SemType semType, MappingAtomicType[] pos, MappingAtomicType[] neg) {
+        this.semType = semType;
+        this.pos = pos;
+        this.neg = neg;
+    }
+
+    public MappingAlternative[] mappingAlternativesRw(Context cx, SemType t) {
+        if (t instanceof UniformTypeBitSet) {
+            if ((((UniformTypeBitSet) t).bitset & PredefinedType.MAPPING_RW.bitset) == 0) {
+                return new MappingAlternative[]{};
+            } else {
+                return new MappingAlternative[]{
+                        from(cx, PredefinedType.MAPPING_RW, new ArrayList<>(), new ArrayList<>())
```

Comment:
Since we are not adding anything to these lists we can use `List.of()` https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/List.html#of()